### PR TITLE
Saving and restoring viewer's last extent. The viewer will now set it's extent to the last extent it saved. On initialization, the viewer will set it's extent over Israel if no extent is saved (will probably happen only the first time the application launches). Saving and restoring is done with Shared-Preferences so it's persistent.

### PR DESCRIPTION
**Pull request by BarAgam**
Saving and restoring viewer's last extent.
The viewer will now set it's extent to the last extent it saved.
On initialization, the viewer will set it's extent over Israel if no extent is saved (will probably happen only the first time the application launches).
Saving and restoring is done with Shared-Preferences so it's persistent.

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 1]
```diff
@@ -59,6 +62,20 @@ public void setGGMapGestureListener(OnMapGestureListener onMapGestureListener) {
     }
 
     @Override
+    public void saveState() {
+        SharedPreferences prefs = getDefaultSharedPreferences();
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:66]: UserPreferenceRepository? Or maybe (my preferred option) just remove it from project.

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 2]
```diff
@@ -59,6 +62,20 @@ public void setGGMapGestureListener(OnMapGestureListener onMapGestureListener) {
...
+        String retainState = retainState();
+        prefs.edit().putString(ESRI_STATE_PREF_KEY, retainState).apply();
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:68]: Inline retainState?

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 3]
```diff
@@ -148,12 +165,35 @@ private boolean isBaseLayerLoaded(Object o, STATUS status) {
...
 
     private void onBasemapLoaded() {
-        setExtentOverIsrael();
+        setExtent();
         addDynamicGraphicLayer();
         notifyMapReady();
         setOnStatusChangedListener(null);
     }
 
+    private void setExtent() {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java]: You can delete this method and add an else clause to restoreState, which calls setExtentOverIsrael.
As for now it's kinda duplicate (restoreState and setExtent shares code)...

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java [hunk 4]
```diff
@@ -148,12 +165,35 @@ private boolean isBaseLayerLoaded(Object o, STATUS status) {
...
+        } else {
+            setExtentOverIsrael();
+        }
+    }
+
+    private boolean hasLastExtent() {
+        return getDefaultSharedPreferences().contains(ESRI_STATE_PREF_KEY);
+    }
+
+    private SharedPreferences getDefaultSharedPreferences() {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java]: Not in place (cascading order)

**Review by teamagam5 — APPROVED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:66]: UserPreferenceRepository is used inside domain (and possibly data, even-though it doesn't have to).
SharedPreferences can be used app (it's even better IMO).

File: app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java (hunk 2 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java:68]: Not inlined to promote the actual data retrieval call

**Pull request closed without merging**